### PR TITLE
Fix wrong handling on reuse class with identifier

### PR DESCRIPTION
**Pull request by chandratjioe**
## Changes in this pull request

Issue fixed: #1176 

### Checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [x] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [x] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by zhubofei** [comment]
@rnystrom Maybe we should add some tests on this one.

File: Source/Internal/IGListAdapterInternal.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ IGListBatchContext
 @property (nonatomic, assign) BOOL isInUpdateBlock;
 @property (nonatomic, strong, nullable) IGListSectionMap *previousSectionMap;
 
-@property (nonatomic, strong) NSMutableSet<Class> *registeredCellClasses;
+@property (nonatomic, strong) NSMutableSet<NSString *> *registeredCellClassIdentifiers;
```
Comment (zhubofei) [inline Source/Internal/IGListAdapterInternal.h]: IMO, `registeredReusableViewIdentifiers` would be a better property name.

File: Source/IGListAdapter.m [hunk 2]
```diff
@@ -977,13 +977,13 @@ - (__kindof UICollectionViewCell *)dequeueReusableCellOfClass:(Class)cellClass
     IGParameterAssert(index >= 0);
     UICollectionView *collectionView = self.collectionView;
     IGAssert(collectionView != nil, @"Dequeueing cell of class %@ with reuseIdentifier %@ from section controller %@ without a collection view at index %li", NSStringFromClass(cellClass), reuseIdentifier, sectionController, (long)index);
-    NSString *identifier = IGListReusableViewIdentifier(cellClass, nil, nil, reuseIdentifier);
+    NSString *classIdentifier = IGListReusableViewIdentifier(cellClass, nil, nil, reuseIdentifier);
```
Comment (zhubofei) [inline Source/IGListAdapter.m]: We can keep using `identifier` here instead of `classIdentifier`.

**Comment by rnystrom** [comment]
I'd love some unit tests for this!

**Comment by rnystrom** [comment]
@chandratjioe do you think you'll be able to add some unit tests?

**Comment by pm-dev** [comment]
Just ran into this issue.
@chandratjioe will you be able to finish this PR? Or perhaps @rnystrom you'd be able to bring this across the finish line? I believe this is the right fix.

**Comment by lorixx** [comment]
Hi @zhubofei Can you rebase again? thanks

**Comment by hanton** [comment]
hi @lorixx , I think this issue has been [fixed](https://github.com/Instagram/IGListKit/blob/cb344f981477d0d8617d8d378ab140e9c8508f70/Source/Internal/IGListAdapterInternal.h#L65) in the latest `master` branch.

**Comment by iperry90** [comment]
Closing this PR since the issue has been fixed.

**Pull request closed without merging**